src/rt/mutex.rs

```diff
@@ -0,0 +1,143 @@
+use crate::rt::object::{self, Object};
+use crate::rt::{thread, Access, Synchronize};
+
+use std::sync::atomic::Ordering::{Acquire, Release};
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub(crate) struct Mutex {
+    obj: Object,
+}
+
+#[derive(Debug)]
+pub(super) struct State {
+    /// If the mutex should establish sequential consistency.
+    seq_cst: bool,
+
+    /// `true` when the mutex is currently in the locked state.
```

Comment:
Nit: I think this comment is no longer accurate?

Reply:
Thanks. I updated the comment.